vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -146,13 +155,15 @@ type internal FSharpProjectOptionsManager
 
     /// Update the info for a project in the project table
     member this.UpdateProjectInfo(tryGetOrCreateProjectId, projectId: ProjectId, site: IProjectSite, workspace: Workspace, userOpName) =
-        this.AddOrUpdateProject(projectId, (fun isRefresh -> 
-            let extraProjectInfo = Some(box workspace)
-            let tryGetOptionsForReferencedProject f = f |> tryGetOrCreateProjectId |> Option.bind this.TryGetOptionsForProject
-            let referencedProjects, options = ProjectSitesAndFiles.GetProjectOptionsForProjectSite(Settings.LanguageServicePerformance.EnableInMemoryCrossProjectReferences, tryGetOptionsForReferencedProject, site, site.ProjectFileName(), extraProjectInfo, serviceProvider, true)
-            let referencedProjectIds = referencedProjects |> Array.choose tryGetOrCreateProjectId
-            checkerProvider.Checker.InvalidateConfiguration(options, startBackgroundCompileIfAlreadySeen = not isRefresh, userOpName= userOpName + ".UpdateProjectInfo")
-            referencedProjectIds, options))
+        async {
+            this.AddOrUpdateProject(projectId, (fun isRefresh -> 
+                let extraProjectInfo = Some(box workspace)
+                let tryGetOptionsForReferencedProject f = f |> tryGetOrCreateProjectId |> Option.bind this.TryGetOptionsForProject
+                let referencedProjects, options = ProjectSitesAndFiles.GetProjectOptionsForProjectSite(Settings.LanguageServicePerformance.EnableInMemoryCrossProjectReferences, tryGetOptionsForReferencedProject, site, site.ProjectFileName(), extraProjectInfo, serviceProvider, true)
+                let referencedProjectIds = referencedProjects |> Array.choose tryGetOrCreateProjectId
+                checkerProvider.Checker.InvalidateConfiguration(options, startBackgroundCompileIfAlreadySeen = not isRefresh, userOpName= userOpName + ".UpdateProjectInfo")
+                referencedProjectIds, options))
+        } |> Async.Start
```

Comment:
This looks like an orthogonal fix.  We need to think about this carefully - the computation of the options may now terminate in different orders, for example, with update A terminating after update B.

I think it's best to put this in a separate PR.